### ISSUE TITLE
Package request: fabric-ai

### ISSUE THREAD

**Issue by Isomorph70** [NixOS/nixpkgs#320031]
**fabric is an open-source framework for augmenting humans using AI.**

Fabric was created to address this by enabling everyone to granularly apply AI to everyday challenges.

fabric is an open-source framework for augmenting humans using AI. It provides a modular framework for solving specific problems using a crowdsourced set of AI prompts that can be used anywhere. 

**Metadata**
ai work augmentation flourishing cli
* homepage URL: https://github.com/danielmiessler/fabric?tab=readme-ov-file#introduction-video-by-network-chuck
* source URL: https://github.com/danielmiessler/fabric
* license: mit.
* platforms: unix, linux, darwin, anything that's runs python

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Isomorph70** [comment]
https://github.com/danielmiessler/fabric/issues/548

**Comment by Isomorph70** [comment]
https://github.com/danielmiessler/fabric/issues/576

**Comment by Charging1948** [comment]
They are currently doing a rewrite in Golang, and are almost done with it. Perhaps a good idea to wait until the Golang-Rewrite is available:

> May 23, 2024 — We will be switching Fabric to Go in a few weeks to avoid all the installation issues with Python. The Go version will be dead-simple to install and will be even faster. Plus easier to update. We already have it working thanks to the heroic efforts of xssdoctor, and we're just working on testing now! Stay tuned for more info on the release date!

Right below Table of Contents in README.md: https://github.com/danielmiessler/fabric

**Comment by Isomorph70** [comment]
pbcopy and pbpaste are helper apps, the functionality can be replicated by adding these lines to 
configuration.nix

environment.interactiveShellInit = ''
  alias pbcopy='xsel --input --clipboard'
  alias pbpaste='xsel --output --clipboard'
'';

**Comment by jaredmontoya** [comment]
they finally switched to go

**Comment by jaredmontoya** [comment]
Package PR: https://github.com/NixOS/nixpkgs/pull/335359

**Comment by Isomorph70** [comment]
Cool. I am going to a solar punk festival in Germany. But I will take a look later.

**Comment by nilp0inter** [comment]
Are there any blockers for PR #335359? It would be nice to have fabric-ai available in nixpkgs.

**Comment by chewblacka** [comment]
> Are there any blockers for PR #335359? It would be nice to have fabric-ai available in nixpkgs.

I think they are waiting on the license to be clarified:

https://github.com/danielmiessler/fabric/issues/833

**Comment by Apocrathia** [comment]
> I think they are waiting on the license to be clarified:
> 
> [danielmiessler/fabric#833](https://github.com/danielmiessler/fabric/issues/833)

Correct. A [PR](https://github.com/danielmiessler/fabric/pull/893) for the license was just submitted and will hopefully be merged soon. There's no reason for this to be marked as unfree.

**Comment by chewblacka** [comment]
Closing as package how now been merged.
